system/lib/jsmath.cpp

```diff
@@ -0,0 +1,56 @@
+#include <emscripten.h>
```

Comment:
Could this file be implement in C rather than C++?

Reply:
Maybe add a comment at the top of this file to say what its for, and how its used?   Mention the JS_MATH option and the separate libjsmath.a library.?

Reply:
Seems so. Changed now.

Why do you prefer C btw?

Reply:
When I suggested it I just thought it seems like a more conservative option.

But now that I think about, it seems pretty important (essential maybe) that libc and libm and libmalloc, etc, don't depend on libc++ or libc++abi.